packages/react-components/lib/task-summary/task-summary-accordion.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+import Debug from 'debug';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import { IconButton, makeStyles, Typography } from '@material-ui/core';
+import { TreeItem, TreeView } from '@material-ui/lab';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+import PlayCircleFilledWhiteIcon from '@material-ui/icons/PlayCircleFilledWhite';
+import PauseCircleFilledIcon from '@material-ui/icons/PauseCircleFilled';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import SimpleInfo, { SimpleInfoData } from '../simple-info';
+import { formatStatus, getActorFromStatus, getStateLabel } from './task-summary-utils';
+
+const debug = Debug('OmniPanel:TaskSummaryAccordion');
+
+interface TaskSummaryAccordionInfoProps {
+  task: RomiCore.TaskSummary;
+}
+
+export const TaskSummaryAccordionInfo = (props: TaskSummaryAccordionInfoProps) => {
+  const { task } = props;
+  const statusDetails = formatStatus(task.status);
+  const stateLabel = getStateLabel(task.state);
+  const data = [
+    { name: 'TaskId', value: task.task_id, wrap: false },
+    { name: 'State', value: stateLabel },
+    { name: 'Status', value: statusDetails },
+    { name: 'Submission Time', value: task.submission_time.sec },
+    { name: 'Start time', value: task.start_time.sec },
+    { name: 'End Time', value: task.end_time.sec },
+  ] as SimpleInfoData[];
+
+  return <SimpleInfo data={data} />;
+};
+
+export interface TaskSummaryAccordionProps {
+  tasks: RomiCore.TaskSummary[];
+}
+
+export const TaskSummaryAccordion = React.memo((props: TaskSummaryAccordionProps) => {
+  debug('task summary status panel render');
+  const { tasks } = props;
+  const classes = useStyles();
+  const [expanded, setExpanded] = React.useState<string[]>([]);
+  const [selected, setSelected] = React.useState<string>('');
+
+  const handleToggle = (event: React.ChangeEvent<{}>, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  const handleSelect = (event: React.ChangeEvent<{}>, nodeIds: string) => {
+    setSelected(nodeIds);
+  };
+
+  // Added a useCallback here because this function is been used inside the useEffect.
+  const determineStyle = (state: number): string => {
+    switch (state) {
+      case RomiCore.TaskSummary.STATE_QUEUED:
+        return classes.queued;
+      case RomiCore.TaskSummary.STATE_ACTIVE:
+        return classes.active;
+      case RomiCore.TaskSummary.STATE_COMPLETED:
+        return classes.completed;
+      case RomiCore.TaskSummary.STATE_FAILED:
+        return classes.failed;
+      default:
+        return 'UNKNOWN';
+    }
+  };
+
+  const renderActor = (taskStatus: string): React.ReactElement | undefined => {
```

Comment:
I don't know when to use `React.ReactElement` or `JSX.Element` I guess both is fine, but I think we should return `null` instead of `undefined`.

Reply:
Cool I think `React.ReactElement` it's ok here

JSX.Element is a ReactElement, with the generic type for props and type being any. It exists, as various libraries can implement JSX in their own way, therefore JSX is a global namespace that then gets set by the library, React sets it like this:

```
declare global {
  namespace JSX {
    interface Element extends React.ReactElement<any, any> { }
  }
}
```


Source https://stackoverflow.com/questions/58123398/when-to-use-jsx-element-vs-reactnode-vs-reactelement